db/db_test2.cc

```diff
@@ -1129,23 +1138,42 @@ TEST_F(DBTest2, DISABLED_PresetCompressionDict) {
       ASSERT_EQ(0, NumTableFilesAtLevel(0, 1));
       ASSERT_GT(NumTableFilesAtLevel(1, 1), 0);
 
-      size_t out_bytes = 0;
+      // Count the live sst files size
+      size_t all_sst_bytes = 0;
       std::vector<std::string> files;
-      GetSstFiles(env_, dbname_, &files);
+      uint64_t manifest_file_size = 0;
+      ASSERT_OK(db_->GetLiveFiles(files, &manifest_file_size));
       for (const auto& file : files) {
-        uint64_t curr_bytes;
-        env_->GetFileSize(dbname_ + "/" + file, &curr_bytes);
-        out_bytes += static_cast<size_t>(curr_bytes);
+        if (file.substr(file.size() - 4, std::string::npos) == ".sst") {
+          uint64_t curr_bytes;
+          env_->GetFileSize(dbname_ + "/" + file, &curr_bytes);
+          all_sst_bytes += static_cast<size_t>(curr_bytes);
+        }
+      }
+      if (i == kWithoutDict) {
+        bytes_without_dict = all_sst_bytes;
+      } else if (i == kWithDict) {
+        bytes_with_dict = all_sst_bytes;
+      } else if (i == kWithZSTDTrainedDict) {
+        bytes_with_zstd_trained_dict = all_sst_bytes;
       }
 
       for (size_t j = 0; j < kNumL0Files * (kL0FileBytes / kBlockSizeBytes);
            j++) {
         ASSERT_EQ(seq_datas[(j / 10) % 10], Get(1, Key(static_cast<int>(j))));
       }
-      if (i) {
-        ASSERT_GT(prev_out_bytes, out_bytes);
+      if (i == kWithDict) {
+        ASSERT_GT(bytes_without_dict, bytes_with_dict);
+      } else if (i == kWithZSTDTrainedDict) {
+        // In zstd compression, it is sometimes possible that using a trained
+        // dictionary does not get as good a compression ratio as without
+        // training.
+        // But using a dictionary (with or without training) should always get
+        // better compression ratio than not using one.
+        ASSERT_TRUE(bytes_with_dict > bytes_with_zstd_trained_dict ||
+                    bytes_without_dict > bytes_with_zstd_trained_dict);
```

Comment:
The condition before || doesn't seem to be very useful.

Reply:
That's the condition from when the test was originally created, and hence I retained it.